store/searchlayer/user_layer.go

```diff
@@ -146,45 +145,48 @@ func (s *SearchUserStore) autocompleteUsersInChannelByEngine(engine searchengine
 	return autocomplete, nil
 }
 
+// getListOfAllowedChannelsForTeam We're contemplating different scenarios in this method:
```

Comment:
actually is more a explanation, something like `getListOfAllowedChannelsForTeam return the list of allowed channels contemplating different scenarios:`